template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hdfs/HdfsRoleConfigProvider.java

```diff
@@ -81,6 +87,20 @@ protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, Templat
         }
     }
 
+    @Override
+    public List<ApiClusterTemplateConfig> getServiceConfigs(CmTemplateProcessor templateProcessor, TemplatePreparationObject source) {
+        List<ApiClusterTemplateConfig> configs = new ArrayList<>();
+        if (isSDXOptimizationEnabled(source)) {
+            if (isNamenodeHA(source)) {
+                configs.add(config(DFS_REPLICATION, DFS_REPLICATION_VALUE.toString()));
+            }
+            configs.add(config(DFS_ENCRYPT_DATA_TRANSFER, "true"));
+            configs.add(config(DFS_DATA_TRANSFER_PROTECTION, "privacy"));
+            configs.add(config(HADOOP_RPC_PROTECTION, "privacy"));
+        }
+        return configs;
+    }
+
     private boolean isSDXOptimizationEnabled(TemplatePreparationObject source) {
```

Comment:
We have 2 different version of this logical switch, one here and the other is in the other config provider?
Can't we have the same logic for both?
`entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId()`